### PR TITLE
Adjust doc links for new scheme

### LINKED ISSUE #8831: Update all links after doc restructuring
Similarly to what 93d2ccbfb7b99755e8f4a2a3947c466949ee3c46 did, we'll have to update all links in the source code to point to the correct `stable/` locations for when 7.0.0 is released (and/or add redirects in RTD's backend).

Otherwise, as I understand it, once the release is out, all the `stable/` links which currently work (due to `stable` pointing to `6.2.x`, before the restructuring) will break :cry:

### PR DESCRIPTION
**Pull request by The-Compiler**
Closes #8831 and supersedes #8830.

Some notes:

1) Since we have redirects in place now (#8471), we don't strictly need to do this (except for a couple of cases where the contents moved, I'll comment on those below). However, I still think it's good if we point to the "canonical" location of the docs.
2) Those links won't work right now, unless you replace `/stable/` by `/latest/`. They will however work once 7.0 is released and RTD regenerates the stable docs.

I'd probably hold off merging this shortly before the 7.0 release (perhaps even the final one rather than the prerelease?) so that we don't break links for people trying the rc or running from git before 7.0 is out.

cc @DrGFreeman

### CODE REVIEW

File: doc/en/changelog.rst [hunk 1]
```diff
@@ -1677,7 +1677,7 @@ Features
 
 - `#1682 <https://github.com/pytest-dev/pytest/issues/1682>`_: The ``scope`` parameter of ``@pytest.fixture`` can now be a callable that receives
   the fixture name and the ``config`` object as keyword-only parameters.
-  See `the docs <https://docs.pytest.org/en/stable/fixture.html#dynamic-scope>`__ for more information.
+  See `the docs <https://docs.pytest.org/en/stable/how-to/fixtures.html#dynamic-scope>`__ for more information.
```
Comment (The-Compiler) [inline doc/en/changelog.rst:1680]: Content moved (`how-to`, not `explanation`).

File: doc/en/example/markers.rst [hunk 2]
```diff
@@ -234,17 +234,17 @@ You can ask which markers exist for your test suite - the list includes our just
...
+    @pytest.mark.parametrize(argnames, argvalues): call a test function multiple times passing in different arguments in turn. argvalues generally needs to be a list of values if argnames specifies only one name or a list of tuples of values if argnames specifies multiple names. Example: @parametrize('arg1', [1,2]) would lead to two calls of the decorated test function, one with arg1=1 and another with arg1=2.see https://docs.pytest.org/en/stable/how-to/parametrize.html for more info and examples.
 
-    @pytest.mark.usefixtures(fixturename1, fixturename2, ...): mark tests as needing all of the specified fixtures. see https://docs.pytest.org/en/stable/fixture.html#usefixtures
+    @pytest.mark.usefixtures(fixturename1, fixturename2, ...): mark tests as needing all of the specified fixtures. see https://docs.pytest.org/en/stable/how-to/fixtures.html#usefixtures
```
Comment (The-Compiler) [inline doc/en/example/markers.rst:247]: Content moved (`how-to`, not `explanation`).

File: doc/en/announce/release-2.9.0.rst [hunk 3]
```diff
@@ -131,7 +131,7 @@ The py.test Development Team
   with same name.
 
 
-.. _`traceback style docs`: https://pytest.org/en/stable/usage.html#modifying-python-traceback-printing
+.. _`traceback style docs`: https://pytest.org/en/stable/how-to/output.html#modifying-python-traceback-printing
```
Comment (The-Compiler) [inline doc/en/announce/release-2.9.0.rst:134]: Content moved

File: doc/en/changelog.rst [hunk 4]
```diff
@@ -2026,7 +2026,7 @@ Features
   `pytest-faulthandler <https://github.com/pytest-dev/pytest-faulthandler>`__ plugin into the core,
   so users should remove that plugin from their requirements if used.
 
-  For more information see the docs: https://docs.pytest.org/en/stable/usage.html#fault-handler
+  For more information see the docs: https://docs.pytest.org/en/stable/how-to/failures.html#fault-handler
```
Comment (The-Compiler) [inline doc/en/changelog.rst:2029]: Content moved

File: doc/en/changelog.rst [hunk 5]
```diff
@@ -4223,7 +4223,7 @@ Features
 
 - Support for Python 3.7's builtin ``breakpoint()`` method, see `Using the
   builtin breakpoint function
-  <https://docs.pytest.org/en/stable/usage.html#breakpoint-builtin>`_ for
+  <https://docs.pytest.org/en/stable/how-to/failures.html#using-the-builtin-breakpoint-function>`_ for
```
Comment (The-Compiler) [inline doc/en/changelog.rst:4226]: Content moved

File: doc/en/changelog.rst [hunk 6]
```diff
@@ -6517,7 +6517,7 @@ time or change existing behaviors in order to make them less surprising/more use
 * Fix (`#1422`_): junit record_xml_property doesn't allow multiple records
   with same name.
 
-.. _`traceback style docs`: https://pytest.org/en/stable/usage.html#modifying-python-traceback-printing
+.. _`traceback style docs`: https://docs.pytest.org/en/stable/how-to/output.html#modifying-python-traceback-printing
```
Comment (The-Compiler) [inline doc/en/changelog.rst:6520]: Content moved

**Review by nicoddemus — APPROVED** [review]
Thanks for pursuing this!

**Comment by The-Compiler** [comment]
There's a reason I said:

> I'd probably hold off merging this shortly before the 7.0 release (perhaps even the final one rather than the prerelease?) so that we don't break links for people trying the rc or running from git before 7.0 is out.

Now we're getting PRs like #8929 undoing those fixes again... :disappointed:

**Pull request merged**